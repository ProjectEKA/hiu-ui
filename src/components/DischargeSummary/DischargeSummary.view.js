import React from 'react';
import * as PropTypes from 'prop-types';
import Typography from '@material-ui/core/Typography';
import Grid from '@material-ui/core/Grid';
import Box from '@material-ui/core/Box';
import useStyle from './DischargeSummary.style';
import { formatDateString } from '../common/HealthInfo/FhirResourcesUtils';

const DischargeSummary = ({ title, startDate, endDate, authors, status, date }) => {
  const classes = useStyle();
  const start = formatDateString(startDate);
  const end = formatDateString(endDate);
  return (
    title ? (
      <Grid container className={classes.container} justify='space-between'>
        <Grid item>
          <Typography variant='h6' color='inherit' className={classes.title}>
            Document: 
            {' '}
            {title}
          </Typography>
        </Grid>
        <Grid item className={classes.description}>
          {date && (
          <div>
            <span className={classes.label}>Date: </span>
            {date}
          </div>
)}
          {start && end && (
          <span className={classes.period}>
            Admitted from
            {start}
            {' '}
            to
            {end}
          </span>
)}
          {authors.length && (
          <div>
            <span className={classes.label}>Authors: </span>
            { authors.join(', ') }
          </div>
)}
          {status && (
          <div>
            <span className={classes.label}>Status: </span>
            {status.toUpperCase()}
          </div>
)}
        </Grid>
      </Grid>
    ) : ''
  );
};

DischargeSummary.propTypes = {
  title: PropTypes.string,
  startDate: PropTypes.string,
  endDate: PropTypes.string,
  authors: PropTypes.arrayOf(PropTypes.string),
  status: PropTypes.string,
  date: PropTypes.string,
};

DischargeSummary.defaultProps = {
  title: '',
  startDate: '',
  endDate: '',
  authors: [],
  status: '',
  date: '',
};


export default DischargeSummary;
