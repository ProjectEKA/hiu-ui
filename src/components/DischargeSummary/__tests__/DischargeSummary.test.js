import React from 'react';
import {shallow} from "enzyme";
import {Typography} from "@material-ui/core";
import DischargeSummary from "../DischargeSummary.view";

describe('DischargeSummary', () => {
    let wrapper;
    global.TIMEZONE_OFFSET = '-05:00';

    beforeEach(() => {
        wrapper = shallow(
          <DischargeSummary
            title="Discharge Summary"
            startDate="2018-04-01T15:30:10+01:00"
            endDate="2018-04-10T15:30:10+01:00"
            authors={['abc', 'xyz']}
            status="final"
          />
        );
    });

    it('should render properly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should have default props', () => {
        expect(DischargeSummary.defaultProps.title).toEqual('');
        expect(DischargeSummary.defaultProps.startDate).toEqual('');
        expect(DischargeSummary.defaultProps.endDate).toEqual('');
        expect(DischargeSummary.defaultProps.authors).toEqual([]);
        expect(DischargeSummary.defaultProps.status).toEqual('');
    });

    it('should not render anything if title is empty', () => {
        wrapper.setProps({title:''});
        expect(wrapper.find(Typography)).toHaveLength(0);
    });

    describe('Period', () => {
        it('should render period if startDate is not empty', () => {
            expect(wrapper.find('span[className*="period"]')).toHaveLength(1);
        });

        it('should not render period if startDate is empty', () => {
            wrapper.setProps({ startDate: '' });
            expect(wrapper.find('span[className*="period"]')).toHaveLength(0);
        });

        it('should render period if endDate is not empty', () => {
            expect(wrapper.find('span[className*="period"]')).toHaveLength(1);
        });

        it('should not render period if endDate is empty', () => {
            wrapper.setProps({ endDate: '' });
            expect(wrapper.find('span[className*="period"]')).toHaveLength(0);
        });
    });

    describe('Authors', () => {
        it('should render author if it is not empty', () => {
            const labels = wrapper.find('span[className*="label"]');
            expect(labels.first().text()).toContain("Authors: ");
        });

        it('should not render author if it is empty', () => {
            wrapper.setProps({ authors: [] });
            const labels = wrapper.find('span[className*="label"]');
            expect(labels.first().text()).not.toContain("Authors: ");
        });
    });

    describe('Status', () => {
        it('should render status if it is not empty', () => {
            const labels = wrapper.find('span[className*="label"]');
            expect(labels.at(1).text()).toContain("Status: ");
        });

        it('should not render status if it is empty', () => {
            wrapper.setProps({ status: '' });
            const labels = wrapper.find('span[className*="label"]');
            expect(labels).toHaveLength(1);
            expect(labels.first().text()).not.toContain("Status: ");
        });
    });
});
