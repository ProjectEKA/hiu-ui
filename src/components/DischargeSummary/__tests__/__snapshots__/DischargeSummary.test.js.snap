// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DischargeSummary should render properly 1`] = `
<WithStyles(ForwardRef(Grid))
  className="makeStyles-container-1"
  container={true}
  justify="space-between"
>
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <WithStyles(ForwardRef(Typography))
      className="makeStyles-title-2"
      color="inherit"
      variant="h6"
    >
      Document:
       
      Discharge Summary
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    className="makeStyles-description-3"
    item={true}
  >
    <span
      className="makeStyles-period-5"
    >
      Admitted from
      01/04/2018
       
      to
      10/04/2018
    </span>
    <div>
      <span
        className="makeStyles-label-4"
      >
        Authors: 
      </span>
      abc, xyz
    </div>
    <div>
      <span
        className="makeStyles-label-4"
      >
        Status: 
      </span>
      FINAL
    </div>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
